Add scenario modification drawing tools

kdeloach opened the pull request:
This adds the ability to add and remove features to the map.

Connects #166

kdeloach commented:
I'm in the process of reverting the changes I made regarding "current conditions".

kdeloach commented:
Rebased and ready for testing.

kdeloach commented:
Rebased onto develop. There's one bug that I'm fixing now. Since each scenario tab has the same slug, switching tabs won't quite work as expected.

caseypt commented:
Checking it out now.

caseypt commented:
Not sure if this needs to happen as part of this PR, but we can persist and reload the modifications now.

caseypt commented:
Another thing that may not be a part of this PR: Clicking on a modification to get info, delete it, etc.

mmcfarland commented:
Already got a separate issue for persisting, unless tossing it in helps to test that things are working.  Though it might be good to focus on whatever nuances come up.  https://github.com/WikiWatershed/model-my-watershed/issues/226

caseypt commented:
I created three scenarios in addition to current conditions, and renamed them so that the tabs work. 

When I draw on the 3rd scenario, the new shape, plus any previous shapes show up on the map. When I switch back to the first scenario, all of the shapes are still on the map. Then, when I switch back to the 3rd scenario, none of the shapes are there.

Let me know if that makes sense.

caseypt commented:
@kdeloach I can still reproduce the above if I update the scenario name through the console, and ensure that the slug has been updated too. However, your method of testing was better, so I may be seeing an unrelated issue.

caseypt commented:
If you think of any tests to add, add them to the list: https://github.com/WikiWatershed/model-my-watershed/issues/242

kdeloach commented:
I'll try to reproduce this issue with the Marionette inspector.

mmcfarland commented on src/mmw/js/src/modeling/views.js:
```diff
@@ -173,16 +179,69 @@ var ScenarioDropDownMenuView = Marionette.CompositeView.extend({
+            featureType = $el.data('feature-type'),
+            map = App.getLeafletMap();
+        this.model.disableControls();
+        drawUtils.drawPolygon(map).then(function(geojson) {
```
Still really like this promise syntax for drawing, nice.

caseypt commented on src/mmw/js/src/modeling/views.js:
```diff
@@ -199,6 +258,8 @@ var ToolbarTabContentsView = Marionette.CollectionView.extend({
 
     onRender: function() {
+        // TODO: Move this logic to the template (active tab should
+        // be stored in model state somewhere).
```
I think I made this comment already, but I can't find it. So here it is again: There's a attribute on project called `active-scenario-slug` that is used to set the active tab. I did it that way since it required less state. If it's more of a pain to have it there, we should add it to scenario, then remove `active-scenario-slug`.

kdeloach replied:
Sorry, it got rebased away. Definitely sounds like "initial scenario" and "active scenario slug" could be combined. I'll think it through a little more.

caseypt replied:
Yeah `initialScenario` is `activeScenarioSlug`, but I thought the name `initialScenario` made more sense as it got passed down and the view was rendered initially. Open to whatever changes though.

kdeloach commented:
Fixed some rebase errors, updated model attributes to use snake case in some places, and removed  obsolete code related to enabling/disabling controls while drawing.

kdeloach commented:
@caseypt Updating these models via Marionette Inspector won't work because we don't presently re-render the tab panels when the scenario model is updated. I'll try to fix this as part of this PR.

See how the tab panels all have the same ID even after I altered the name/slug of each scenario.

![image](https://cloud.githubusercontent.com/assets/43062/7891255/6cb8226c-0619-11e5-9473-fb964559eb4f.png)

kdeloach commented:
Fixed all the tab problems (that I'm aware of). Everything should be working now.

caseypt commented:
+1. Previously reported issue re: tab switching and persisting shapes has been fixed. I can give it another run through when you rebase it on develop.

kdeloach commented:
Rebased.

caseypt commented:
Tested again. Looks good.

caseypt commented on src/mmw/js/src/modeling/models.js:
```diff
@@ -18,6 +18,16 @@ var ResultCollection = Backbone.Collection.extend({
+        model_package: null,       // ModelPackageModel
+        taskModel: null,           // TaskModel
+        scenarios: null,           // ScenariosCollection
+        active_scenario_slug: ''
```
I think we can kill this now that we have `active` on scenario?

kdeloach replied:
Removed.

caseypt commented on src/mmw/js/src/modeling/views.js:
```diff
@@ -228,6 +272,10 @@ var ModelingResultsWindow = Marionette.LayoutView.extend({
     tagName: 'div',
     template: resultsWindowTmpl,
 
+    modelEvents: {
```
Sorry, here too.

kdeloach replied:
Thanks, will remove in a follow up PR.
